### DISCUSSION TITLE
URock Processor Log Error

### DISCUSSION THREAD

**Discussion by AlexKDo** [UMEP-dev/UMEP#514] [Q&A]
Hi, I am trying to run URock but I have an error Here is the log.

QGIS version: 3.30.1-'s-Hertogenbosch
QGIS code revision: 9035a01e
Qt version: 5.15.3
Python version: 3.9.5
GDAL version: 3.6.3
GEOS version: 3.11.2-CAPI-1.17.2
PROJ version: Rel. 9.2.0, March 1st, 2023
PDAL version: 2.5.2 (git-version: 57c4e7)
Algorithm started at: 2023-04-09T13:55:14
Algorithm 'Urban Wind Field: URock v2023a' starting…
Input parameters:
{ 'ATTENUATION_FIELD' : '', 'BUILDINGS' : 'D:/ΠΜΣ/ΜΕΤΑΠΤΥΧΙΑΚΗ ΔΙΑΤΡΙΒΗ/My research and conclusions/ExperimentFiles/raw.data/scenario_00_basic/URockPrepare/BuildingFile.shp', 'HEIGHT_FIELD_BUILD' : 'ROOF_HEIGH', 'HORIZONTAL_RESOLUTION' : 2, 'INPUT_PROFILE_FILE' : '', 'INPUT_PROFILE_TYPE' : 0, 'INPUT_WIND_DIRECTION' : 45, 'INPUT_WIND_HEIGHT' : 10, 'INPUT_WIND_SPEED' : 2, 'JAVA_PATH' : 'C:\\Program Files\\Java\\jre1.8.0_361', 'LOAD_OUTPUT' : True, 'OUTPUT_FILENAME' : 'urock_output', 'RASTER_OUTPUT' : None, 'SAVE_NETCDF' : True, 'SAVE_RASTER' : True, 'SAVE_VECTOR' : True, 'UROCK_OUTPUT' : 'D:\\ΠΜΣ\\ΜΕΤΑΠΤΥΧΙΑΚΗ ΔΙΑΤΡΙΒΗ\\My research and conclusions\\ExperimentFiles\\raw.data\\scenario_00_basic\\URockProcessor', 'VEGETATION' : 'D:/ΠΜΣ/ΜΕΤΑΠΤΥΧΙΑΚΗ ΔΙΑΤΡΙΒΗ/My research and conclusions/ExperimentFiles/raw.data/scenario_00_basic/URockPrepare/VegetationFile.shp', 'VEGETATION_CROWN_BASE_HEIGHT' : '', 'VEGETATION_CROWN_TOP_HEIGHT' : 'VEG_HEIGHT', 'VERTICAL_RESOLUTION' : 2, 'WIND_HEIGHT' : '1.5' }

Writing settings for this model run to specified output folder (Filename: RunInfoURock_YYYY_DOY_HHMM.txt)
Initiating algorithm
Creates an H2GIS Instance and load data
Creates the stacked blocks used as obstacles
Rotates obstacles to the right direction and calculates geometry properties
Creates the 2D Röckle zones
Creates the 2D grid
Initializes the 3D grid within Röckle zones
Deals with building zones superimposition
Deals with vegetation zones superimposition
Initialize the 3D wind in the grid
Rasterize the data
Apply the wind solver equations
Iteration 1 (max 500) - eps = 0.606513 >= 0.0001

Iteration 51 (max 500) - eps = 0.011632 >= 0.0001

Iteration 101 (max 500) - eps = 0.002362 >= 0.0001

Iteration 151 (max 500) - eps = 0.000481 >= 0.0001

Traceback (most recent call last):
File "JdbcPreparedStatement.java", line 230, in org.h2.jdbc.JdbcPreparedStatement.execute
java.lang.java.lang.IllegalArgumentException: java.lang.IllegalArgumentException: No transformation found from epsg:2100 to epsg:4326

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
File "JdbcPreparedStatement.java", line 230, in org.h2.jdbc.JdbcPreparedStatement.execute
Exception: Java Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
File "C:\OSGeo4W\apps\Python39\lib\site-packages\jaydebeapi\__init__.py", line 534, in execute
is_rs = self._prep.execute()
org.h2.jdbc.org.h2.jdbc.JdbcSQLNonTransientException: org.h2.jdbc.JdbcSQLNonTransientException: Exception calling user-defined function: "ST_Transform(conn241: url=jdbc:default:connection user=SA, POINT (475283.4855100708 4207015.674958611), 4326): No transformation found from epsg:2100 to epsg:4326"; SQL statement:

SELECT ST_X(THE_GEOM) AS LON, ST_Y(THE_GEOM) AS LAT FROM 
(SELECT ST_TRANSFORM(ST_SETSRID(THE_GEOM,2100), 4326) AS THE_GEOM FROM GRID_ROTATED)
[90105-200]

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
File "C:\Users/alexk/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\processing_umep\processor\urock_processing_algorithm.py", line 436, in processAlgorithm
MainCalculation.main(javaEnvironmentPath = javaEnvVar,
File "C:\Users/alexk/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\processing_umep\functions\URock\MainCalculation.py", line 774, in main
saveData.saveBasicOutputs(cursor = cursor , z_out = z_out,
File "C:\Users/alexk/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\processing_umep\functions\URock\saveData.py", line 42, in saveBasicOutputs
cursor.execute("""
File "C:\OSGeo4W\apps\Python39\lib\site-packages\jaydebeapi\__init__.py", line 536, in execute
_handle_sql_exception()
File "C:\OSGeo4W\apps\Python39\lib\site-packages\jaydebeapi\__init__.py", line 165, in _handle_sql_exception_jpype
reraise(exc_type, exc_info[1], exc_info[2])
File "C:\OSGeo4W\apps\Python39\lib\site-packages\jaydebeapi\__init__.py", line 57, in reraise
raise value.with_traceback(tb)
File "C:\OSGeo4W\apps\Python39\lib\site-packages\jaydebeapi\__init__.py", line 534, in execute
is_rs = self._prep.execute()
jaydebeapi.DatabaseError: org.h2.jdbc.JdbcSQLNonTransientException: Exception calling user-defined function: "ST_Transform(conn241: url=jdbc:default:connection user=SA, POINT (475283.4855100708 4207015.674958611), 4326): No transformation found from epsg:2100 to epsg:4326"; SQL statement:

SELECT ST_X(THE_GEOM) AS LON, ST_Y(THE_GEOM) AS LAT FROM 
(SELECT ST_TRANSFORM(ST_SETSRID(THE_GEOM,2100), 4326) AS THE_GEOM FROM GRID_ROTATED)
[90105-200]

Execution failed after 68.93 seconds (1 minute 9 seconds)

Loading resulting layers
Algorithm 'Urban Wind Field: URock v2023a' finished

**Comment by biglimp** [comment]
It seems like there is a problem with CRS. You use a national Greek CRS which URock cannot translate to WGS84. Try reproject your data to a UTM e.g. ED50 / UTM zone 34N and ED50 / UTM zone 35N

**Comment by j3r3m1** [comment]
I might add a test at the beginning of the URock module to check whether URock will be able to convert a CRS to 4326. If not then I might try converting the CRS to the corresponding UTM but I need to find the way to do that automatically in QGIS

**Comment by j3r3m1** [comment]
OK I have forwarded the problem to the maintainers of the H2GIS database, they should solve the problem soon. 

See https://github.com/orbisgis/h2gis/issues/1341
Thanks for reporting

**Comment by ebocher** [comment]
This bug has been fixed on the current H2GIS snapshot version 2.2.1 
See https://github.com/orbisgis/h2gis/pull/1342

**Reply by j3r3m1** [reply]
Excellent, thanks ! Next version of UMEP will then include this version of H2GIS

**Comment by j3r3m1** [comment]
@AlexKDo can you please retry with the last UMEP version (download the zip file [here](https://github.com/UMEP-dev/UMEP-processing/archive/refs/heads/main.zip) ?

Once downloaded, you should import directly the zip into QGIS in the "Plugin" -> "Manage and install Plugins" -> "Install from zip".
Reload QGIS if needed.

**Comment by AlexKDo** [comment]
I reinstalled from zip and run the Göteborg test dataset and I have an error again

QGIS version: 3.32.3-Lima
QGIS code revision: 67d46100b5
Qt version: 5.15.3
Python version: 3.9.5
GDAL version: 3.7.2
GEOS version: 3.12.0-CAPI-1.18.0
PROJ version: Rel. 9.3.0, September 1st, 2023
PDAL version: 2.5.5 (git-version: 24f78d)
Algorithm started at: 2024-06-08T14:16:09
Algorithm 'Urban Wind Field: URock v2023a' starting…
Input parameters:
{ 'ATTENUATION_FIELD' : '', 'BUILDINGS' : 'C:\\Users\\ALEXAN~1\\AppData\\Local\\Temp\\build_vector.GeoJSON', 'HEIGHT_FIELD_BUILD' : 'ROOF_HEIGHT', 'HORIZONTAL_RESOLUTION' : 2, 'INPUT_PROFILE_FILE' : '', 'INPUT_PROFILE_TYPE' : 0, 'INPUT_WIND_DIRECTION' : 45, 'INPUT_WIND_HEIGHT' : 10, 'INPUT_WIND_SPEED' : 2, 'LOAD_OUTPUT' : True, 'OUTPUT_FILENAME' : 'urock_output', 'RASTER_OUTPUT' : None, 'SAVE_NETCDF' : True, 'SAVE_RASTER' : True, 'SAVE_VECTOR' : True, 'UROCK_OUTPUT' : 'TEMPORARY_OUTPUT', 'VEGETATION' : 'C:\\Users\\ALEXAN~1\\AppData\\Local\\Temp\\veg_vector.GeoJSON', 'VEGETATION_CROWN_BASE_HEIGHT' : '', 'VEGETATION_CROWN_TOP_HEIGHT' : 'VEG_HEIGHT', 'VERTICAL_RESOLUTION' : 2, 'WIND_HEIGHT' : '1.5' }

Traceback (most recent call last):
File "C:\Users/Alexandra/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\UMEP-processing-main\functions\URock\H2gisConnection.py", line 349, in getJavaHome
java_key = winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE, r"SOFTWARE\JavaSoft\Java Runtime Environment")
FileNotFoundError: [WinError 2] Δεν είναι δυνατή η εύρεση του καθορισμένου αρχείου από το σύστημα

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
File "C:\Users/Alexandra/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\UMEP-processing-main\functions\URock\H2gisConnection.py", line 353, in getJavaHome
java_key = winreg.OpenKey(winreg.HKEY_LOCAL_MACHINE, r"SOFTWARE\JavaSoft\Java Development Kit")
FileNotFoundError: [WinError 2] Δεν είναι δυνατή η εύρεση του καθορισμένου αρχείου από το σύστημα

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
File "C:\Users/Alexandra/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\UMEP-processing-main\processor\urock_processing_algorithm.py", line 324, in processAlgorithm
javaDirDefault = getJavaDir(plugin_directory)
File "C:\Users/Alexandra/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\UMEP-processing-main\functions\URock\H2gisConnection.py", line 234, in getJavaDir
javaPath = getJavaHome(os_type)
File "C:\Users/Alexandra/AppData/Roaming/QGIS/QGIS3\profiles\default/python/plugins\UMEP-processing-main\functions\URock\H2gisConnection.py", line 356, in getJavaHome
exit()
File "C:\PROGRA~1\QGIS33~1.2\apps\Python39\lib\_sitebuiltins.py", line 26, in __call__
raise SystemExit(code)
SystemExit: None

Execution failed after 0.02 seconds

Loading resulting layers
Algorithm 'Urban Wind Field: URock v2023a' finished

**Comment by j3r3m1** [comment]
Which version of Java do you have ? Can you give us the result of a "java -version" ?
